docs/manual/repository-management.md

```diff
@@ -0,0 +1,210 @@
+# Managing content in OSTree repositories
+
+Once you have a build system going, if you actually want client
+systems to retrieve the content, you will quickly feel a need for
+"repository management".
+
+OSTree itself does not currently come with tools to do this.  One
+reason is that how content is delivered and managed has concerns very
+specific to the organization.  For example, some operating system
+content vendors may want integration with a specific errata
+notification system.
+
+In this section, we will describe some high level ideas and methods
+for managing content in OSTree repositories, mostly independent of any
+particular model or tool.  That said, a goal is to include at least
+some sample scripts and workflows upstream in a potential new
+"contrib" git repository.
+
+Today, the [Pulp Project](http://www.pulpproject.org/) which is a
+generic content management system, does have an
+[Pulp OSTree plugin](https://pulp-ostree.readthedocs.org/en/latest/).
+
+## Separate development vs release repositories
+
+By default, OSTree accumulates server side history.  This is actually
+optional in that your build system can (using the API) write a commit
+with no parent.  But first, we'll investigate the ramifications of
+server side history.
+
+Many content vendors will want to separate their internal development
+with what is made public to the world.  Therefore, you will want (at
+least) two OSTree repositories, we'll call them "dev" and "prod".
+
+To phase this another way, let's say you have continuous delivery
+system which is building from git and committing into your "dev"
+OSTree repository.  This might happen tens to hundreds of times per
+day.  That's a substantial amount of history over time, and it's
+unlikely most of your content consumers (i.e. not developers/testers)
+will be interested in all of it.
+
+The original vision of OSTree was to fulfill this "dev" role, and in
+particular the "archive-z2" format was designed for it.
+
+Then, what you'll want to do is promote content from "dev" to "prod".
+We'll discuss this later, but first, let's talk about promotion
+*inside* our "dev" repository.
+
+## Promoting content along OSTree branches - "buildmaster", "smoketested"
+
+Besides multiple repositories, like git, OSTree also supports multiple
```

Comment:
Nit: I would put ", like git" at the end of the sentence, otherwise it parses as relating to the opening clause.